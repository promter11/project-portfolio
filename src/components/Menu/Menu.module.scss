@import "../../styles/_variables";

.nav {
  display: flex;
  flex-direction: column;
  margin: 0 -4rem;

  @media (max-width: 992px) {
    margin: 0 -3rem;
  }

  .link {
    display: flex;
    align-items: center;
    padding: 1.5rem 4rem 1.5rem 3.6rem;
    border-left: 0.4rem solid transparent;
    transition: all 0.2s ease-in-out;

    @media (max-width: 992px) {
      justify-content: center;
      padding: 1.5rem 2rem;
    }

    &.active {
      border-left: 0.4rem solid $white-color;
      background-color: $link-background-color-active;
    }

    &:hover {
      &:not(.active) {
        border-left: 0.4rem solid $white-color;
        background-color: $link-background-color-hover;
      }
    }
  }

  .icon {
    width: 2rem;
    height: 2rem;
    margin-right: 3rem;

    @media (max-width: 992px) {
      margin-right: 0;
    }
  }

  .text {
    font-size: 1.4rem;
    font-weight: bold;
    line-height: 100%;
    text-transform: uppercase;
    color: $white-color;

    @media (max-width: 992px) {
      display: none;
    }
  }
}
