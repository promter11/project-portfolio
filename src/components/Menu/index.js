import React from "react";
import { string, func, arrayOf, shape } from "prop-types";
import { NavLink } from "react-router-dom";

import styles from "./Menu.module.scss";

const Menu = ({ items, onToggleStatus }) => {
  return (
    <nav className={styles.nav}>
      {items.map(({ title, path, icon }, index) => {
        return (
          <NavLink
            key={index}
            className={styles.link}
            activeClassName={styles.active}
            to={path}
            exact
            onClick={() => onToggleStatus()}
          >
            <img className={styles.icon} src={icon} alt={title} />
            <span className={styles.text}>{title}</span>
          </NavLink>
        );
      })}
    </nav>
  );
};

Menu.propTypes = {
  items: arrayOf(
    shape({
      title: string,
      path: string,
      icon: string,
    })
  ),
  onToggleStatus: func,
};

export default Menu;
