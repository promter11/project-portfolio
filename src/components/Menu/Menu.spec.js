import React from "react";
import { shallow } from "enzyme";

import Menu from "./index";

const props = {
  items: [
    { title: "Home", path: "/", icon: "../icons/menu/1" },
    { title: "About", path: "/about", icon: "../icons/menu/2" },
  ],
  onToggleStatus: jest.fn(),
};

describe("Menu component", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Menu {...props} />);
  });

  it("renders correctly with props", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("simulates onClick event on onToggleStatus function", () => {
    wrapper.find("NavLink").at(0).simulate("click");

    expect(props.onToggleStatus).toHaveBeenCalled();
  });
});
