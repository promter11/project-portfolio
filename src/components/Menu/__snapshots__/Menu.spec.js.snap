// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu component renders correctly with props 1`] = `
<nav
  className="nav"
>
  <NavLink
    activeClassName="active"
    className="link"
    exact={true}
    key="0"
    onClick={[Function]}
    to="/"
  >
    <img
      alt="Home"
      className="icon"
      src="../icons/menu/1"
    />
    <span
      className="text"
    >
      Home
    </span>
  </NavLink>
  <NavLink
    activeClassName="active"
    className="link"
    exact={true}
    key="1"
    onClick={[Function]}
    to="/about"
  >
    <img
      alt="About"
      className="icon"
      src="../icons/menu/2"
    />
    <span
      className="text"
    >
      About
    </span>
  </NavLink>
</nav>
`;
